### PR TITLE
add Play-Annotations project. prepare Scala 3

### PR DESCRIPTION
**Pull request by xuwei-k**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Purpose

add new sub project and class for workaround Scala 3 issue.

## Background Context

- https://github.com/lampepfl/dotty/issues/12840

## References

- https://github.com/playframework/playframework/issues/11260

### CODE REVIEW

File: project/BuildSettings.scala [hunk 1]
```diff
@@ -195,6 +195,7 @@ object BuildSettings {
       (organization.value %% moduleName.value % version).cross(cross)
     }.toSet,
     mimaBinaryIssueFilters ++= Seq(
+      ProblemFilters.exclude[MissingClassProblem]("play.inject.NamedImpl"),
```
[the remark this answers is not in the document]
Comment (octonato) [reply project/BuildSettings.scala:198]: Akka has a few cases where we have different source code depending on the Scala version being used during compilation. 

For example: https://github.com/akka/akka/blob/main/build.sbt#L106-L113 and  https://github.com/akka/akka/tree/main/akka-actor/src/main

[the remark this answers is not in the document]
Comment (xuwei-k) [reply project/BuildSettings.scala:198]: - https://github.com/playframework/playframework/pull/11337 🤔

**Comment by jtjeferreira** [comment]
https://github.com/lampepfl/dotty/issues/12840 was fixed, but scala 3.3.0-RC1 was not yet released...

**Comment by mkurz** [comment]
3.3.0-RC1 released last week: https://github.com/lampepfl/dotty/releases/tag/3.3.0-RC1
@xuwei-k Can you rebase please, thanks!

**Comment by mkurz** [comment]
(Or probably we don't even need this PR anymore?)

**Comment by jtjeferreira** [comment]
I think this can be closed

**Comment by AlexITC** [comment]
Finally, 3.3.0-RC2 is out https://github.com/lampepfl/dotty/releases/tag/3.3.0-RC2

**Pull request closed without merging**